manim/__main__.py

```diff
@@ -1,14 +1,178 @@
-from . import extract_scene
-from . import config
+import inspect
+import itertools as it
+import os
+import platform
+import subprocess as sp
+import sys
+import traceback
+import importlib.util
+
+from .config import file_writer_config
+from .scene.scene import Scene
+from .utils.sounds import play_error_sound
+from .utils.sounds import play_finish_sound
 from . import constants
+from .logger import logger
+
+
+def open_file_if_needed(file_writer):
+    if file_writer_config["quiet"]:
+        curr_stdout = sys.stdout
+        sys.stdout = open(os.devnull, "w")
+
+    open_file = any([
+        file_writer_config["preview"],
+        file_writer_config["show_file_in_finder"]
+    ])
+    if open_file:
+        current_os = platform.system()
+        file_paths = []
+
+        if file_writer_config["save_last_frame"]:
+            file_paths.append(file_writer.get_image_file_path())
+        if file_writer_config["write_to_movie"]:
+            file_paths.append(file_writer.get_movie_file_path())
+
+        for file_path in file_paths:
+            if current_os == "Windows":
+                os.startfile(file_path)
+            else:
+                commands = []
+                if current_os == "Linux":
+                    commands.append("xdg-open")
+                elif current_os.startswith("CYGWIN"):
+                    commands.append("cygstart")
+                else:  # Assume macOS
+                    commands.append("open")
+
+                if file_writer_config["show_file_in_finder"]:
+                    commands.append("-R")
+
+                commands.append(file_path)
+
+                # commands.append("-g")
+                FNULL = open(os.devnull, 'w')
+                sp.call(commands, stdout=FNULL, stderr=sp.STDOUT)
+                FNULL.close()
+
+    if file_writer_config["quiet"]:
+        sys.stdout.close()
+        sys.stdout = curr_stdout
+
+
+def is_child_scene(obj, module):
+    if not inspect.isclass(obj):
+        return False
+    if not issubclass(obj, Scene):
+        return False
+    if obj == Scene:
+        return False
+    if not obj.__module__.startswith(module.__name__):
+        return False
+    return True
+
+
+def prompt_user_for_choice(scene_classes):
+    num_to_class = {}
+    for count, scene_class in zip(it.count(1), scene_classes):
+        name = scene_class.__name__
+        print("%d: %s" % (count, name))
+        num_to_class[count] = scene_class
```

Comment:
i think this would be more idiomatic 
```suggestion
    for count, scene_class in enumerate(scene_classes):
        count += 1  # start with 1 instead of 0
        name = scene_class.__name__
        print("%d: %s" % (count, name))
        num_to_class[count] = scene_class
```